File: quickstarts/VB/EventBroker/Tests/Tests.EventBrokerExtension/SimpleEventBrokerExtensionFixture.vb
' Copyright (c) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.

Imports EventBrokerExtension
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports Microsoft.Practices.Unity
Imports SimpleEventBroker
Imports Tests.EventBrokerExtension.Tests.EventBrokerExtension.TestObjects

<TestClass()> _
Public Class SimpleEventBrokerExtensionFixture
    <TestMethod()> _
    Public Sub ContainerCanWireEvents()
        Dim container As IUnityContainer = New UnityContainer() _
                .AddNewExtension (Of SimpleEventBrokerExtension)()

        Dim clipboard As ClipboardManager = container.Resolve (Of ClipboardManager)()

        Dim broker As EventBroker = container.Configure (Of ISimpleEventBrokerConfiguration)().Broker

        Dim registeredEvents As New List(Of String) (broker.RegisteredEvents)
        registeredEvents.Sort()

        Dim expectedEvents As New List(Of String) (New String() {"cut", "copy", "paste", "clipboard data available"})
        expectedEvents.Sort()

        CollectionAssert.AreEqual (expectedEvents, registeredEvents)
    End Sub
End Class

File: quickstarts/CS/EventBroker/Tests/Tests.EventBrokerExtension/SimpleEventBrokerExtensionFixture.cs
// Copyright (c) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.

using System.Collections.Generic;
using EventBrokerExtension;
using Microsoft.Practices.Unity;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using SimpleEventBroker;
using Tests.EventBrokerExtension.TestObjects;

namespace Tests.EventBrokerExtension
{
    [TestClass]
    public class SimpleEventBrokerExtensionFixture
    {
        [TestMethod]
        public void ContainerCanWireEvents()
        {
            IUnityContainer container = new UnityContainer()
                .AddNewExtension<SimpleEventBrokerExtension>();

            ClipboardManager clipboard = container.Resolve<ClipboardManager>();

            EventBroker broker = container.Configure<ISimpleEventBrokerConfiguration>().Broker;

            List<string> registeredEvents = new List<string>(broker.RegisteredEvents);
            registeredEvents.Sort();

            List<string> expectedEvents = new List<string>(new string[]
            {
                "cut",
                "copy",
                "paste",
                "clipboard data available"
            });
            expectedEvents.Sort();

            CollectionAssert.AreEqual(expectedEvents, registeredEvents);
        }
    }
}
